School.js

```diff
@@ -1,71 +1,104 @@
+      physics1: {
+        name: "physics1",
+        teacher: "mr.stkrdiknmibalz",
+        students: [
+          {
+            name: "stan marsh",
+            city: "south park",
+            age: 7,
+            grade: 70
+          },
+          {
+            name: "eric cartman",
+            city: "south park",
+            age: 7,
+            grade: 60
+          },
+          {
+            name: "kenny mccormick",
+            city: "south park",
+            age: 8,
+            grade: 100
+          },
+          {
+            name: "leopold stoch",
+            city: "south park",
+            age: 7,
+            grade: 75
+          },
+          {
+            name: "wendy testaburger",
+            city: "south park",
+            age: 7,
+            grade: 99
+          },
+          {
+            name: "kirk brovloski",
+            city: "south park",
+            age: 9,
+            grade: 99
+          }
+        ]
+      }
     }
   }
 
-  /**
-   * Add class to classes
-   * 
-   * @param {string} name - Name of the class
-   * @param {string} teacher - Name of instructor 
-   * @return {Class} Class object
-   */
   addClass(name, teacher) {
-    let newClass = new Class(name, teacher);
-    this.classes[name] = newClass;
+    let newClass = new Class(name, teacher)
+    this.classes[name] = newClass
   }
 
-  /**
-   * Enroll student in class
-   * 
-   * @param {string} className - Name of the class
-   * @param {Student} student - Student object
-   * @return {Student} Enrolled student
-   */
-  enrollStudent(className, student) {
-    // Your code here
+  enrollStudent(className, stdnt) {
+    this.classes[className].addStudent(stdnt)
+    return stdnt
   }
 
-
-
-
-  /**
-   * Get all students enrolled in a class
-   * 
-   * @param {string} className - Name of the class
-   * @return {Student[]} Array of Student objects
-   */
   getStudentsByClass(className) {
-    // Your code here
+    return this.classes[className].students
   }
 
-
-
-
-  /**
-   * Get all students and apply filters. If failing = true
-   * return all students that are failing the class, 
-   * that is all students whose grade is less than 70.
-   * If a city is passed return students whose city match
-   * the city passed. If both failing and city are passed
-   * return students that are failing and that live in the
-   * specified city
-   * 
-   * @param {string} className - Name of the class
-   * @param {boolean} failing - Whether to return students that are failing the class or not
-   * @param {string} city - Name of the city to match against students
-   * @return {Student[]} Array of Student objects
-   */
   getStudentsByClassWithFilter(className, failing, city) {
     // Your code here
+    let arr = mySchool.classes[className].students
+    console.log(arr)
+    if (city !== undefined) {
+      arr = arr.filter(s => s.city === city)
+      console.log(s.city)
+    }
+    if (failing !== undefined && failing === "true") {
```

Comment:
this would overwrite your the previous result is city was undefined, it would be better this was a `else if`